python/src/spark_rapids_ml/knn.py

```diff
@@ -221,6 +247,31 @@ class NearestNeighbors(
     |{2, [3.0, 3.0]}|{4, [3.0, 3.0]}|              0.0|
     |{1, [2.0, 2.0]}|{4, [3.0, 3.0]}|        1.4142135|
     +---------------+---------------+-----------------+
+
+    >>> from spark_rapids_ml.knn import NearestNeighbors
+    >>> from pyspark.ml.linalg import Vectors
+    >>> data = [(0, Vectors.dense([1.0, 1.0]),),
+    ...         (1, Vectors.dense([2.0, 2.0]),),
+    ...         (2, Vectors.dense([3.0, 3.0]),)]
+    >>> data_df = spark.createDataFrame(data, ["id", "features"])
+    >>> query = [(3, Vectors.dense([1.0, 1.0]),),
+    ...          (4, Vectors.dense([3.0, 3.0]),)]
+    >>> query_df = spark.createDataFrame(query, ["id", "features"])
+    >>> topk = 2
+    >>> gpu_knn = NearestNeighbors().setInputCol("features").setIdCol("id").setK(topk)
+    >>> gpu_model = gpu_knn.fit(data_df)
+
+    >>> from spark_rapids_ml.knn import NearestNeighbors
```

Comment:
I would remove this second import on the assumption that this is a single session.

Reply:
And maybe add comment lines explaining the two varieties, e.g.
```
# single column input
...
# multi-column input
```

Reply:
did not delete the second import, thinking the example codes may be copied to a different session. 

array input is also single column so added the comments "vector column input" and "multi-column input".